### DISCUSSION TITLE
No Nightly builds from July 25th

### DISCUSSION THREAD

**Discussion by tsunamistate** [winsiderss/systeminformer#957] [Questions / Answers] [answered]
Hello!
I really like this project and have a Nightly build installed, but there are no new builds since f3f6ced1006befef2a1390ee2bbd53de440d7362 even though there were new commits since then

@dmex did you disable CI for your repo? If you did, why?

**Comment by dmex** [comment, marked as answer]
> did you disable CI for your repo?

No? Nothing has changed. It might be confusing when compared to other projects but I've always triggered the CI build manually and it has never been automatic. I configured it this way to prevent github and/or CI itself from being an attack vector and to guarantee higher quality nightly build releases.

So your question should be why haven't I triggered the CI for the nightly builds? Two big changes required more thorough testing than usual.

* I've added a new restart menu using the undocumented Windows BCD (boot configuration data) functions allowing you to reboot the machine directly to various options, devices and operating systems:

"Reboot to firmware options"
"Reboot to boot application"
"Reboot to firmware application"

These three menus let you reboot directly from the desktop into the BIOS setup screen or reboot directly into EFI applications (e.g EFIShell and Memory Diagnostics) and/or EFI devices (DVD/Ethernet/USB devices) and separate operating systems (even directly into a VHD file if you've configured one).

![image](https://user-images.githubusercontent.com/1306177/130396330-64d287f8-13c2-4a65-9840-df0b435cd73d.png)

![image](https://user-images.githubusercontent.com/1306177/130396451-0e7b0b97-e9d6-48c1-883c-b982f4e6221e.png)

Process Hacker is the first application to use the undocumented BCD functions to a large extent and the first application to include boot loader reboot functionality such as this feature. 

I want to be absolutely sure it doesn't cause any unintended side effects or other issues like failed startups or boot loops... To that end, we're using the 'one-time' feature provided by BCD for these restarts, so failed configurations are guaranteed to be automatically recovered. That feature needs to work, so I've been running tests and asking a few people for code reviews before releasing the feature to everyone on the nightly branch.

TLDR: I want the reboot feature fully-baked rather than half-baked. 

* The second major change was including support for inline stack traces and changes to imports for minidump support:

![image](https://user-images.githubusercontent.com/1306177/130409768-453ff343-a3e3-48b9-b62c-e375e629e3d2.png)

You can see the new (inline function) entries on the thread stack as well as stack source line information. This completely changed how PH enumerated thread stacks as well as changed code required for MiniDumpWriteDump which is used for both the Process > Create dump feature but also our own exception handling dialog. For example: 

![image](https://user-images.githubusercontent.com/1306177/130412417-44de16f8-9197-4356-abae-62a8cdd682b2.png)

I don't want to be in the position where the new boot configuration data functionality causes a crash but we don't ever know about it because our new stack support doesn't capture the crash information or somehow broke the exception handling.

TLDR: Brand new reboot feature + brand new exception/stalk walking/minidump support = accident waiting to happen.

> there were new commits since then

Yep. I've been committing other changes as well but there were also some commits fixing and improving both of those features. So it was a good idea to pause nightly releases and let them 'bake' a bit longer so I can guarantee they're working properly.

Everything is looking good so I'll be releasing both the new reboot to device/reboot to firmware feature and the new dump support/inline stack features in the next few days and resuming the nightly builds 👍

**Reply by tsunamistate** [reply]
I didn't know that you trigger nightly builds manually, so I thought CI broke and wanted to notify you of that.
As for new features, they are exciting, especially "boot to firmware"! Can't wait until it's ready for prime time ;) 
Thank you for working on PH! This software really helps me in monitoring my PC, it beats the pants off default Task Manager.

**Reply by dmex** [reply]
> Can't wait until it's ready

The nightly builds have been updated with the new features 👍
